compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1004,25 +1004,49 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
     fn lower_assoc_ty_constraint(
         &mut self,
         constraint: &AssocTyConstraint,
-        itctx: ImplTraitContext<'_, 'hir>,
+        mut itctx: ImplTraitContext<'_, 'hir>,
```

Comment:
We need `reborrow` calls when passing `itctx` to `lower_angle_bracket_parameter_data`, because we need `itctx` later when we pass it to `lower_ty`. The `reborrow`s after the `lower_angle_bracketed_parameter_data` call are unnecessary though. I've removed them.